### ISSUE TITLE
vSync limits the fps to 80fps on macOS

### ISSUE THREAD

**Issue by warrex1979** [yquake2/yquake2#745]
I was just playing the 8.00 release on macOS Big Sur 11.6 (Mac Mini 2020 with M1 SoC connceted to a Sony KD-48A9 OLED TV) and noticed that even when vid_maxfps and cl_maxfps are both set to 120 vSync limits the fps to 80 (OpenGL 1.4 and 3.2). Turning it off results in an absolutely stable 120fps@1080p. I did not encounter this behaviour with the OpenGL backend of GZDoom.

Btw.: Thank you for compiling builds for macOS! It would be nice if you could cross compile for Apple Silicon though.

**Comment by Yamagi** [comment]
The MacOS is maintained by the community. Neither @DanielGibson not me have a mac, so we're unable to help here. If someone is able to figure out whats wrong and sends a pull request with a fix we're happy to merge it.

**Comment by devnexen** [comment]
might depend on the gzdoom settings tough, I have same fps limitations on my mac on both gzdoom and yquake2 on my mac.

For the cross compiling part, theoretically a separate build for arm64 then for x86_64 (rosetta) on a mac m1, then "gluing" them together with lipo to make an universal binary would be ideal.

**Comment by warrex1979** [comment]
> might depend on the gzdoom settings tough, I have same fps limitations on my mac on both gzdoom and yquake2.

At least in GZDoom 4.7.0 I cannot reproduce this. I tested OpenGL, OpenGL ES and Vulkan with VSync on. All of them are locked to 120fps if I set the target fps respectively in the menu.

Thank you for considering cross-compiling (https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary).

Please make sure to sign and notarize the release then (the current release of QuakeSpasm crashes on M1 Macs because the team did not follow Apple's guides):

"New in macOS 11 on Macs with Apple silicon, and starting in macOS Big Sur 11 beta 6, the operating system enforces that any executable must be signed before it’s allowed to run."
https://developer.apple.com/documentation/macos-release-notes/macos-big-sur-11_0_1-universal-apps-release-notes

Code Signing hub page on Apple.com:
https://developer.apple.com/support/code-signing/

Notarizing macOS Software Before Distribution: https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution

**Comment by devnexen** [comment]
I do not know if a paying apple license developer is necessary for the signing part, if that's the case maybe just proposing separate arm64 and x86_64 versions would be more "pragmatic".

**Comment by devnexen** [comment]
That gives enough hints for the mac packager even tough I would not make it sound like "high priority" I am sure it works more than decently in rosetta mode for the vast majority of m1 users.

**Comment by warrex1979** [comment]
Running natively on Apple Silicon can be much faster or efficent. Here is an GZDoom example with a level that is infamous for the way it stresses the engine / CPU: https://forum.zdoom.org/viewtopic.php?t=71303#p1179875

**Comment by tomkidd** [comment]
Does anyone have it building for Apple Silicon yet? 

I just got done helping ioquake3 do a Universal 2 build (Intel 64-bit and Apple Silicon) with signing and notarization ( https://ioquake3.org/ioquake3/legacy-and-m1-macos-ioquake3-support/ ) and I have just done the same with vkQuake ( https://www.dropbox.com/s/5tbw8flru6d6j7m/vkquake_notarized.zip?dl=0 ) so naturally Quake 2 is next. 

Just tried a make of the latest code and it's failing because the makefile is looking at the x86_64 SDL include files versus the arm64 Homebrew location (in my case). I can figure out a way to make both happy just checking first to make sure the wheel hasn't been invented here yet.

**Comment by devnexen** [comment]
Yes I do all the time when I in the terminal for arm64 or rosetta/x86_64 mode my compiler/linker flags are set accordingly (you can have both homebrew in same time).

**Comment by warrex1979** [comment]
Thanks! I am currently away from my M1 but tried to launch it via chrome remote desktop where the the regular build works (strangely @120fps...) . Your build did not launch for whatever reason. No launch menu and not the main app after having copied the ressources. I will be back at the weekend to see if this changes anything.

**Comment by shinra-electric** [comment]
Brilliant, I was looking for this....

I'm getting 618 frames over 5.2 seconds on the M1 Pro, which is 119.8 fps (at 1440p).

The funny thing is that this is what I get at any setting (OGL 1.4, OGL 3.2, anisotropic filtering/multisampling off or max), except for using the software renderer. 

There seems to be an artificial limit of 120fps...? Would I be right in saying that?

**Comment by warrex1979** [comment]
Did you make sure that vsync is off?

**Comment by shinra-electric** [comment]
Holy shit, what was that? 

V-sync off, 793fps

**Comment by warrex1979** [comment]
The M1 Pro's GPU is twice as fast as the M1's GPU and you are running at twice the resolution without using a remote desktop solution as I did. So the fps you are getting seem to be in line with what one should expect.

Even the M1 should be able to run at 4K 120fps - even through Rosetta 2 but for future proofing and running as efficent as possible it is still important to compile for Apple Silicon.

Keep vsync off and set vid_maxfps to 120 (https://github.com/yquake2/yquake2/blob/master/doc/030_configuration.md).
